### PR TITLE
swim bug fixed - image activation for all the child sites devices - V2.3.7.6

### PR DESCRIPTION
**Pull request by syed-khadeerahmed**
## Description
image activation for all the child sites devices - V2.3.7.6

## Type of Change
- [x] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Documentation update

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] All the sanity checks have been completed and the sanity test cases have been executed

## Ansible Best Practices
- [x] Tasks are idempotent (can be run multiple times without changing state)
- [ ] Variables and secrets are handled securely (e.g., using `ansible-vault` or environment variables)
- [x] Playbooks are modular and reusable
- [ ] Handlers are used for actions that need to run on change

## Documentation
- [x] All options and parameters are documented clearly.
- [x] Examples are provided and tested.
- [x] Notes and limitations are clearly stated.

### CODE REVIEW

File: plugins/modules/swim_workflow_manager.py [hunk 1]
```diff
@@ -937,62 +937,60 @@ def get_device_uuids(self, site_name, device_family, device_role, device_series_
...
-                    family="site_design",
-                    function='get_site_assigned_network_devices',
-                    op_modifies=True,
-                    params={"site_id": site_id},
-                )
-                self.log("Received API response from 'get_site_assigned_network_devices': {0}".format(str(response)), "DEBUG")
-                response = response.get('response')
+            site_names = site_name + ".*"
+            get_site_names = self.get_site(site_names)
+            self.log(get_site_names)
```
Comment (madhansansel) [inline plugins/modules/swim_workflow_manager.py:942]: self.log("Fetched site names: {0}".format(str(get_site_names)), "DEBUG")

File: plugins/modules/swim_workflow_manager.py [hunk 2]
```diff
@@ -937,62 +937,60 @@ def get_device_uuids(self, site_name, device_family, device_role, device_series_
...
+            site_info = {}
+
+            for item in get_site_names['response']:
+                if 'nameHierarchy' in item and 'id' in item:
+                    site_info[item['nameHierarchy']] = item['id']
+
```
Comment (madhansansel) [inline plugins/modules/swim_workflow_manager.py:948]:
```
device_id_list = []  # List to store all device IDs
site_response_list = []  # List to store all relevant device responses

```

File: plugins/modules/swim_workflow_manager.py [hunk 3]
```diff
@@ -937,62 +937,60 @@ def get_device_uuids(self, site_name, device_family, device_role, device_series_
...
+            for site_name, site_id in site_info.items():
+                try:
+                    response = self.dnac._exec(
+                        family="site_design",
+                        function='get_site_assigned_network_devices',
+                        op_modifies=True,
```
Comment (madhansansel) [inline plugins/modules/swim_workflow_manager.py:954]: True or False?

File: plugins/modules/swim_workflow_manager.py [hunk 4]
```diff
@@ -937,62 +937,60 @@ def get_device_uuids(self, site_name, device_family, device_role, device_series_
...
+                        params={"site_id": site_id},
+                    )
+                    self.log("Received API response from 'get_site_assigned_network_devices': {0}".format(str(response)), "DEBUG")
+                    response = response.get('response')
```
Comment (madhansansel) [inline plugins/modules/swim_workflow_manager.py:958]: instead:
```
        devices = response.get('response')
        if not devices:
            self.log("No devices found for site - '{0}'.".format(site_name), "WARNING")
            continue
```

File: plugins/modules/swim_workflow_manager.py [hunk 5]
```diff
@@ -937,62 +937,60 @@ def get_device_uuids(self, site_name, device_family, device_role, device_series_
...
 
-                if not response:
-                    self.log("No devices found for site '{0}'.". format(site_name), "WARNING")
+                    if not response:
+                        self.log("No devices found for site - '{0}'.". format(site_name), "WARNING")
 
-            except Exception as e:
-                self.log("Unable to fetch the device(s) associated to the site '{0}' due to '{1}'".format(site_name, str(e)), "WARNING")
-                return device_uuid_list
+                    for device_id in response:
```
Comment (madhansansel) [inline plugins/modules/swim_workflow_manager.py:963]: for device in devices:
            device_id_list.append(device.get("deviceId"))

File: plugins/modules/swim_workflow_manager.py [hunk 6]
```diff
@@ -937,62 +937,60 @@ def get_device_uuids(self, site_name, device_family, device_role, device_series_
...
-                        initial_exec = True
-                        device_list_response = self.dnac._exec(
-                            family="devices",
-                            function='get_device_list',
-                            op_modifies=True
-                        )
-                    offset = offset + 1
-                    self.log("Received API response from 'device_list_response': {0}".format(str(device_list_response)), "DEBUG")
-                    device_response = device_list_response.get('response')
+                    self.log("Received API response: {0}".format(str(device_list_response)), "DEBUG")
```
Comment (madhansansel) [inline plugins/modules/swim_workflow_manager.py:979]: self.log("Received API response from 'get_device_list': {0}".format(str(device_list_response)), "DEBUG")

File: plugins/modules/swim_workflow_manager.py (hunk 2 above)
Comment (syed-khadeerahmed) [reply plugins/modules/swim_workflow_manager.py:949]: I have initialize the both empty list at the top

File: plugins/modules/swim_workflow_manager.py (hunk 1 above)
Comment (syed-khadeerahmed) [reply plugins/modules/swim_workflow_manager.py]: fixed the code

File: plugins/modules/swim_workflow_manager.py (hunk 3 above)
Comment (syed-khadeerahmed) [reply plugins/modules/swim_workflow_manager.py]: fixed the code

Comment (syed-khadeerahmed) [reply plugins/modules/swim_workflow_manager.py]: have removed the op_modifies as it is not required for get

File: plugins/modules/swim_workflow_manager.py (hunk 5 above)
Comment (syed-khadeerahmed) [reply plugins/modules/swim_workflow_manager.py]: fixed the code

File: plugins/modules/swim_workflow_manager.py (hunk 6 above)
Comment (syed-khadeerahmed) [reply plugins/modules/swim_workflow_manager.py]: fixed the code

**Review by madhansansel — APPROVED** [review]

**Pull request merged**